src/model/core/co2.jl

```diff
@@ -134,7 +134,10 @@ function co2!(EP::Model, inputs::Dict)
     end
 
     # emissions by zone
+    RESOURCES_BY_ZONE = map(1:Z) do z
+        return resources_in_zone_by_rid(gen, z)
+    end
     @expression(EP, eEmissionsByZone[z = 1:Z, t = 1:T],
-        sum(eEmissionsByPlant[y, t] for y in resources_in_zone_by_rid(gen, z)))
+        sum(eEmissionsByPlant[y, t] for y in RESOURCES_BY_ZONE[z]))
```

Comment:
`resources_in_zone_by_rid` is:
```julia
function resources_in_zone_by_rid(rs::Vector{<:AbstractResource}, zone::Int)
    return resource_id.(rs[zone_id.(rs) .== zone])
end
```
This was being called for every `z` (needed) and every `t` (not needed). When `T` is large, this is expensive.